skimage/feature/orb.py

```diff
@@ -186,8 +186,8 @@ def detect(self, image):
 
             keypoints_list.append(keypoints * self.downscale ** octave)
             orientations_list.append(orientations)
-            scales_list.append(self.downscale ** octave
-                               * np.ones(keypoints.shape[0], dtype=np.intp))
+            scales_list.append(np.full(
+                keypoints.shape[0], self.downscale ** octave, dtype=np.intp))
```

Comment:
It looks like this is causing a test failure: the unwanted copy, in this case, was also casting the result to float! I think changing dtype to float should fix things!

Other than that, thanks for a great PR! I found out about `np.full` quite late in the game, so it'll be great that people looking at our code will see the proper NumPythonic way of doing things! =)

Reply:
Thanks for the review @jni ! Fixed it, also specified a few dtypes manually as for numpy <1.12, np.full by default used the float64 dtype (same as np.ones), while after it uses the dtype of the second argument instead. So specifying it manually when it's not a float is safer..